gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -401,15 +401,18 @@ private Path prepareJobInput(Path jobInputPath, List<WorkUnit> workUnits) throws
       int multiTaskIdSequence = 0;
       // Serialize each work unit into a file named after the task ID
       for (WorkUnit workUnit : workUnits) {
+
+        Path workUnitFileExt;
```

Comment:
You don't need even to make a `Path`, a `String` is sufficient. Also please rename it to `workUnitFileName`.